gateway/src/service.rs

```diff
@@ -1023,6 +1040,73 @@ impl DockerContext for GatewayContext {
     fn container_settings(&self) -> &ContainerSettings {
         &self.settings
     }
+
+    #[instrument(name = "getting container stats from the proper source", skip_all, fields(docker_stats_source = %self.docker_stats_source, shuttle.container.id = container_id))]
```

Comment:
Just a small nitpick left for me :)

```suggestion
    #[instrument(name = "get container stats", skip_all, fields(docker_stats_source = %self.docker_stats_source, shuttle.container.id = container_id))]
```